connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorPluginsResourceTest.java

```diff
@@ -127,7 +143,42 @@ public void setUp() throws NoSuchMethodException {
     }
 
     @Test
-    public void testValidateConfig() throws Throwable {
+    public void testValidateConfigWithMissingConnectorClassname() throws Throwable {
+        herder.validateConnectorConfig(EasyMock.eq(partialProps));
+
+        PowerMock.expectLastCall().andAnswer(new IAnswer<ConfigInfos>() {
```

Comment:
Git shows this as completely new, but this is not correct. Actually this is the old test, basically with just the name changes. Everything with prefix `partial` was introduced to keep the old behavior. 

I agree that this is confusing and not intuitive, but I didn't want to remove an old test before we make sure we test for the desired behaviour. It doesn't throw an exception because of these two lines: 

```java
String includedConnType = connectorConfig.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG);
        if (includedConnType != null && !includedConnType.endsWith(connType))
```
In that case, `includedConnType` is `null`. That's the old behaviour. You suggest this is a bug?